tests/filecheck/transforms/set-memory-space.mlir

```diff
@@ -0,0 +1,42 @@
+// RUN: ./compiler/snax-opt %s -p set-memory-space --allow-unregistered-dialect --print-op-generic | filecheck %s
+
+"builtin.module"() ({
```

Comment:
I feel like this test is quite complex, since it tests many things in sequence, do you think it would make sense to split up the tests?